scripts/globals/magic_utils/spell_damage.lua

```diff
@@ -0,0 +1,790 @@
+    local modBurst                        = 1.0
+    local ancientMagicBurstBonus          = 0
+    local skillchainTier, skillchainCount = FormMagicBurst(spellElement, target) -- External function. Not present in magic.lua.
+
+    if spellId >= 204 and spellId <= 215 then -- If spell is Ancient Magic
+        ancientMagicBurstBonus = caster:getMerit(xi.merit.ANCIENT_MAGIC_BURST_DMG) / 100
+    end
+
+    -- MBB = 1.0 + Gear + Atma/Atmacite + AMII Merits + others -- This Caps at 1.4
+    -- MBB = MBB + trait
+
+    if spell:getSpellGroup() == 3 and not caster:hasStatusEffect(xi.effect.BURST_AFFINITY) then
+        return magicBurstBonus
+    end
+
+    -- Obtain multiplier from gear, atma and job traits -- Job traits should be done separately
+    modBurst = modBurst + (caster:getMod(xi.mod.MAG_BURST_BONUS) / 100) + ancientMagicBurstBonus
+
+    -- Apply Innin bonus
+    if caster:isBehind(target) and caster:hasStatusEffect(xi.effect.INNIN) then
+        modBurst = modBurst + (caster:getMerit(xi.merit.INNIN_EFFECT) / 100)
+    end
+
+    -- BLM Job Point: Magic Burst Damage
+    modBurst = modBurst + (caster:getJobPointLevel(xi.jp.MAGIC_BURST_DMG_BONUS) / 100)
+
+    -- Cap bonuses from first multiplier at 40% or 1.4
+    if modBurst > 1.4 then
+        modBurst = 1.4
+    end
+
+    -- TODO: BLM job trait has to be separate from gear trait, since the job trait is NOT capped. At least, cap is not known.
+    -- Magic Burst Damage I is found in gear. caps at 40% with innin, AM2 and such
+    -- Magic Burst Damage II is found in other gear and BLM job traits pertain to this one OR to a third, separate one. neither has known cap
+
+    if skillchainCount > 0 then
+        magicBurstBonus = modBurst -- + modBurstTrait once investigated. Probably needs to be divided by 100
+    end
+
+    return magicBurstBonus
+end
+
+xi.magic_utils.spell_damage.calculateDayAndWeather = function(caster, target, spell, spellId, spellElement)
+    local dayAndWeather  = 1 -- The variable we want to calculate
+    local weather        = caster:getWeather()
+    local dayElement     = VanadielDayElement()
+    local isHelixSpell   = false -- TODO: I'm not sure thats the correct way to handle helixes. This is how we handle it and im not gonna change it for now.
+
+    -- See if its a Helix type spell
+    if spellId >= 278 and spellId <= 285 then
+        isHelixSpell = true
+    end
+
+    -- Calculate Weather bonus
+    if weather == xi.magic.singleWeatherStrong[spellElement] then
+        if caster:getMod(xi.mod.IRIDESCENCE) >= 1 then
+            if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell then
+                dayAndWeather = dayAndWeather + 0.10
+            end
+        end
+
+        if math.random() < 0.33 or caster:getMod(elementalObi[spellElement]) >= 1 or isHelixSpell then
+            dayAndWeather = dayAndWeather + 0.10
+        end
+
+    elseif weather == xi.magic.singleWeatherWeak[spellElement] then
```

Comment:
possibly to table/loop much if this if/else tower? it looks like a lot of repeating code goes on here.